### ISSUE TITLE
train loss doesn't decrease while finetuning using training script

### ISSUE THREAD

**Issue by emily-swatchon** [huggingface/diffusers#3604]
Hello, first of all thank you so much for all your contribution.

I am using the newly uploaded [train_text_to_image.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image.py) to fine-tune the text-to-image stable diffusion model on my own dataset using 1 A100 GPU.

However, **the train loss does not decrease** but rather stays the same with some fluctuations for 15K steps, as the tensorboard graph shows below.
<img width="633" alt="image" src="https://github.com/huggingface/diffusers/assets/100738598/ad52b8e9-ed64-4376-bbcf-ba86d860d8b3">

It seems like **the result images through validation prompts show a proof of training** (as it gets closer to the distribution of my own training dataset at later training epochs). 
But I am concerned why the train loss is not decreasing as I expected.

Just to validate that my custom dataset was not the issue, I have tried training with the given Pokemon example using the given command (same batch size, learning rate, etc.) and the train loss didn't decrease either.

Is this an expected behavior? 
Or is it due to inappropriate hyperparameters or some other issue?

Thank you.

**Comment by sayakpaul** [comment]
The training loss is known to be unstable for these use cases. 

I would suggest running validation inference as you're training to ensure the model is learning as expected:

https://github.com/huggingface/diffusers/blob/4f14b363297cf8deac3e88a3bf31f59880ac8a96/examples/text_to_image/train_text_to_image.py#L64

You can also try out the [Min-SNR strategy](https://github.com/huggingface/diffusers/tree/main/examples/text_to_image#training-with-min-snr-weighting) to see if that helps stabilizing the loss surface.

**Comment by WindVChen** [comment]
Hi @sayakpaul ,

Thanks for your advice. Recently I also encountered the problem above, and I am very curious about why we can get good results (or in other words why the model does learn from the data) even if the loss doesn't decrease? And what possibly lead to the not-decreasing loss? Could you explain further what lead to the phenomena?

Hope to hear from you. Thanks.

**Comment by sayakpaul** [comment]
I am closing this issue as it's not directly related to the codebase of diffusers. 

Feel free to reopen in case something pops up that is directly to the diffusers codebase.

**Comment by zhuzhuzhu001** [comment]
I met the same problem,
